### ISSUE TITLE
Master window size/position

### ISSUE THREAD

**Issue by GoogleCodeExporter** [brockgr/csshx#16]
```
As of the latest version of csshX, the master window overlaps with the bottom 
row of slave 
windows, which makes it impossible to read the last line in these windows.
The area used by the slave windows should be resized to accommodate all windows.
```

Original issue reported on code.google.com by `dario.ta...@gmail.com` on 14 Jul 2009 at 2:00

**Comment by GoogleCodeExporter** [comment]
```
Interesting, not seen this.

What resolution are you running at, and how many windows? 

If possible, could you attach a screen shot of this?
```

Original comment by `gavin.brock` on 29 Jul 2009 at 4:01

**Comment by GoogleCodeExporter** [comment]
```
> What resolution are you running at, and how many windows? 
1440x900 (default on Macbook Pro 15")

> If possible, could you attach a screen shot of this?
done - see how part of the scrollbar controls for the bottom row are hidden by 
the master window
```

Original comment by `dario.ta...@gmail.com` on 29 Jul 2009 at 2:20

Attachments:
- [Picture 1.png](https://storage.googleapis.com/google-code-attachments/csshx/issue-16/comment-2/Picture 1.png)

**Comment by GoogleCodeExporter** [comment]
```
Here's my own as I've experienced this issue too.

1680x1050 on an external monitor connected to my macbook pro.

Same problem, except I showed how it munges the bottom line, which is pretty 
paralyzing after just a few short 
commands...
I cropped it down to the relevant chunk, but if you have any questions I should 
be able to answer them.
```

Original comment by `vxjaso...@gmail.com` on 29 Jul 2009 at 8:08

Attachments:
- [csshX Overlap.png](https://storage.googleapis.com/google-code-attachments/csshx/issue-16/comment-3/csshX Overlap.png)

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the shots - makes it clear what the problem is.

I think this may be related to the font size. What font do you use for the 
terminals? I seem to be able to replicate 
it using Monaco 11+ (I usually use 9pt)

The problem is that Leopard won't let a terminal be less than 5 rows high. I 
guess I will need to resize the the 
master window and then check what size it really is before retiling the rest 
(or maybe try to push the master off 
the bottom of the screen)
```

Original comment by `gavin.brock` on 29 Jul 2009 at 10:59
- Changed state: **Accepted**
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
Added code to SVN to check for a master window that is larger than I asked it 
to be
(this only happens on 10.5, 10.4 terminal will do what you say).

If the master is too large I shift it down (maybe off the bottom of the screen).

This also makes the "master_height" in csshrc more useful. You can use it to 
'hide'
the bottom of the master window if you are using full screen bounds.

If you have a chance, could you test: 
http://csshx.googlecode.com/svn/trunk/csshX
```

Original comment by `gavin.brock` on 30 Jul 2009 at 2:06

**Comment by GoogleCodeExporter** [comment]
Original comment by `gavin.brock` on 30 Jul 2009 at 1:18
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
tried the latest from trunk, works great!
```

Original comment by `dario.ta...@gmail.com` on 30 Jul 2009 at 5:34

**Comment by GoogleCodeExporter** [comment]
```
as of the latest version, I cannot re-enable disabled windows!
```

Original comment by `dario.ta...@gmail.com` on 31 Jul 2009 at 3:56

**Comment by GoogleCodeExporter** [comment]
```
I made quite a lot of changes on Friday. Double check you are on the latest are 
on csshX v0.63-80

In what circumstances can you not re-enable? Ctrl-a n, or selecting another 
window? One of the late 63-70's 
version had a problem with the enabled/disabled font colors..
```

Original comment by `gavin.brock` on 1 Aug 2009 at 1:55

**Comment by GoogleCodeExporter** [comment]
```
I installed the latest from trunk. If I Ctrl-a e, disable a window and then try 
to re-enable it, the window works 
fine but the font color still looks as if it was disabled, which is confusing.
```

Original comment by `dario.ta...@gmail.com` on 3 Aug 2009 at 10:39

**Comment by GoogleCodeExporter** [comment]
```
Interestingly enough, I think this issue is resolved automatically in Snow 
Leopard. I currently have a 236x4 
master window, and no window overlap.

I'm currently running the release 0.63, not from SVN.
```

Original comment by `vxjaso...@gmail.com` on 11 Sep 2009 at 4:19

**Comment by GoogleCodeExporter** [comment]
```
Hi Gavin/all

I also found this when running OS X 10.5, I manually fixed csshX adding 15 
pixels (or so) to the variable $y 
right after it does the math to get Y value for the master window... it worked 
like a charm for me...

After upgrading to 10.6, csshX wasn't working any longer so I installed it from 
MacPorts and also failed so I 
applied the workaround described at 
http://code.google.com/p/csshx/issues/detail?id=21 and now is back 
working and master window positioning seems to be correct. (version is 0.63).

Thanks for your work Gavin, this app makes my life easier and I love it... keep 
up the good work!

Juan
```

Original comment by `mor...@gmail.com` on 12 Sep 2009 at 10:28

**Comment by GoogleCodeExporter** [comment]
```
Released in 0.64: http://gavcode.wordpress.com/2010/03/30/csshx-0-64/
```

Original comment by `gavin.brock` on 30 Mar 2010 at 2:05
- Changed state: **Done**
